code/modules/projectiles/self_charge.dm

```diff
@@ -0,0 +1,53 @@
+/obj/item/device/self_charge_module
+	name = "weapon recharging module"
+	desc = "A small device attached to the power cell of an energy weapon, designed to allow it to continue operation when away from a convenient recharger. This one seems outdated, with an abysmally slow recharge rate."
+	icon = 'icons/obj/charge_modules.dmi'
+	icon_state = "low-selfchargemodule"
+	origin_tech = list(TECH_POWER = 3, TECH_COMBAT = 3)
+	flags = CONDUCT
+	w_class = ITEMSIZE_TINY
+	drop_sound = 'sound/items/drop/component.ogg'
+	pickup_sound = 'sound/items/pickup/component.ogg'
+	var/obj/item/gun/energy/gun
```

Comment:
You need to define a Destroy proc where you null this reference if the module is destroyed

Reply:
addressed